mlflow/models/evaluation/default_evaluator.py

```diff
@@ -370,6 +370,33 @@ def __validate_artifacts(artifacts):
     )
 
 
+def _compute_df_mode_or_mean(df):
+    """
+    Compute mode (for integer or non-numeric columns or mean (for float columns) for the
+    input dataframe, return a dict, key is column name, value is the corresponding mode or
+    mean value.
+    """
+    # convert pandas dataframe columns to be the best type.
+    # e.g. if one column has float type values, but they are actually all integers,
+    # it will be converted to integer column
+    df = df.convert_dtypes()
+    columns = df.columns.tolist()
+    float_columns = []
+    non_float_column = []
+    for col_name in columns:
+        if df.dtypes[col_name].name.lower().startswith("float"):
+            float_columns.append(col_name)
+        else:
+            non_float_column.append(col_name)
+
+    means = df[float_columns].mean().to_dict()
+    modes = df[non_float_column].mode().loc[0].to_dict()
+    return {**means, **modes}
+
+
+_SUPPORTED_SHAP_ALGORITHMS = ["exact", "permutation", "partition", "kernel"]
```

Comment:
```suggestion
_SUPPORTED_SHAP_ALGORITHMS = ("exact", "permutation", "partition", "kernel")
```

Let's use a tuple here.